src/main/java/io/vertx/core/json/JsonObject.java

```diff
@@ -88,6 +88,26 @@ public JsonObject(Buffer buf) {
     }
   }
 
+  /**
+   * Create a JsonObject from some mapping pairs from arguments.
+   * Ignore the last argument if the number of arguments is not even.
+   *
+   * @param args The key/value pairs.
+   */
+  public static JsonObject of(Object ...args) {
```

Comment:
Map.of has signatures like:

Map<K, V> of(K k1, V v1, K k2, V v2, K k3, V v3, K k4, V v4, K k5, V v5,
                               K k6, V v6, K k7, V v7, K k8, V v8, K k9, V v9, K k10, V v10);

in our case it would give something similar but with K=String and V=Object instead.

The benefit is that it is more type safe (for the keys) and the argument count is always even.

The drawback is to have a noisy API, but JDK did it.

Shouldn't we follow the same pattern here ?

Reply:
I think we should stick to what JDK do here, if somebody has more than this number of arguments then he should use the other way (i.e create an empty json object and add everything)

Reply:
Ok, I could change that if so. Below is some other information about it before I do make the change.

I thought again about it, and then I remembered another reason of why I didn't use the JDK `Map.of` style.
The Map<K,V> has a `K,V` type but our JsonObject does not like that. Here the `K` is always `String` and `V` is always uncertain and finally be a `Object`. The benefits of the noisy style is less than the `Map.of`.

In fact `Map.of` will create a `Map1` or `MapN`, and the constructor of `MapN` is `MapN(Object... input)` style:
https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/ImmutableCollections.java#L1172

In summary, we could have options of the style:
1. current `JsonObject.of(Object... args)` style
2. Map.of(K k1, V v1...) style, like `JsonObject.of(String k1, Object, v1, String v2, Object, v2...)`
3. Or just remove the `of`, create another constructor JsonObject(Object... args)

Which do you prefer? @vietj 
Also looking forward to your suggestions @pmlopes @tsegismont

Reply:
I think we should use 2. 

I think we want to avoid constructors since we already have the JsonObject(Map) that is often misleading ppl and we should discourage using it.

Reply:
Ok, let me do that then, thanks!